jung-visualization/src/main/java/edu/uci/ics/jung/visualization/subLayout/TreeCollapser.java

```diff
@@ -42,13 +42,23 @@ public static MutableCTreeNetwork collapse(MutableCTreeNetwork tree, Object subR
   }
 
   @SuppressWarnings({"unchecked", "rawtypes"})
-  public static void expand(MutableCTreeNetwork tree, CTreeNetwork subTree) {
+  public static MutableCTreeNetwork expand(MutableCTreeNetwork tree, MutableCTreeNetwork subTree) {
     Optional parent = tree.predecessor(subTree);
     Object parentEdge =
         parent.isPresent()
-            ? tree.inEdges(subTree).iterator().next() // THERE CAN BE ONLY ONE
+            ? Iterables.getOnlyElement(tree.inEdges(subTree)) // THERE CAN BE ONLY ONE
             : null;
     tree.removeNode(subTree);
```

Comment:
this should be moved after the check for whether the subtree root is present

Reply:
Nice catch! Done. :)